### PR TITLE
complete false & true more generally as vals

### LINKED ISSUE #49892: REPL: `true` and `false` aren't autocompleted in function arguments
```
julia> f(x) = x
f (generic function with 1 method)

julia> f(fals
```
press <kbd>TAB</kbd> after `fals` and you get `falses`, instead of `false`.  This happens only in function arguments, `false` and `true` are normally autocompleted when they are outside of function arguments.

### PR DESCRIPTION
**Pull request by IanButterworth**
Fixes https://github.com/JuliaLang/julia/issues/49892

I'm not sure whether separating the concepts of "keyvals" for `false` & `true` from the list of "keywords" is strictly correct, but they seem to be different conceptually.

### CODE REVIEW

File: stdlib/REPL/src/REPLCompletions.jl [hunk 1]
```diff
@@ -926,6 +937,7 @@ function complete_keyword_argument(partial, last_idx, context_module)
 
     suggestions = Completion[KeywordArgumentCompletion(kwarg) for kwarg in kwargs]
     append!(suggestions, complete_symbol(nothing, last_word, Returns(true), context_module))
+    append!(suggestions, complete_keyval(last_word))
```
Comment (IanButterworth) [inline stdlib/REPL/src/REPLCompletions.jl:940]: This is the actual fix

File: stdlib/REPL/src/REPLCompletions.jl [hunk 2]
```diff
@@ -949,6 +961,7 @@ end
 function complete_identifiers!(suggestions::Vector{Completion}, @nospecialize(ffunc::Function), context_module::Module, string::String, name::String, pos::Int, dotpos::Int, startpos::Int, comp_keywords=false)
     ex = nothing
     comp_keywords && append!(suggestions, complete_keyword(name))
+    append!(suggestions, complete_keyval(name))
```
Comment (IanButterworth) [inline stdlib/REPL/src/REPLCompletions.jl:964]: This just seems logical to do. I'm not aware of anything this fixes

Comment (IanButterworth) [reply stdlib/REPL/src/REPLCompletions.jl]: Ok there are failures. I'm trying making this a NFC

**Review by giordano — APPROVED** [review]
I'm not familiar with REPL internals, but I can confirm this PR fixes the reported problem, thanks!  I'm also not sure why `false` and `true` were in `sorted_keywords` together with the other keyword, it doesn't feel like they fit together with them

**Comment by IanButterworth** [comment]
Agreed. I do think this change is logical, and if this does break anything, there weren't tests for it, and it can be tweaked.

**Pull request merged**